### PR TITLE
Searchconsole data bigquery

### PR DESCRIPTION
**Pull request by mufaddaltaiyeb**
Hi @maxious ,

Can you please review and add the script to the repository. The script extracts and loads Google Console Search data into BigQuery.

with Best Regards,
Mufaddal

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -3,5 +3,5 @@
     "python.linting.pycodestyleEnabled": false,
     "python.linting.enabled": true,
     "python.linting.flake8Enabled": true,
-    "python.pythonPath": "/usr/local/opt/python@2/bin/python2.7"
+    "python.pythonPath": "/Users/mufaddal/.pyenv/versions/3.8.2/bin/python"
```
Comment (maxious) [inline .vscode/settings.json:6]: If this is going to change on each machine, maybe this file should be removed and put in gitignore?

File: google_searchconsole/gsc_bq_dta_dga.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+def get_sc_df(site_url, start_row, end_date=datetime.date.today(), days=90):
+    """Grab Search Console data for the specific property and send it to BigQuery."""
+
+    start_date = (end_date - datetime.timedelta(days=days)
+                  ).strftime('%Y-%m-%d')
+    end_date = end_date.strftime('%Y-%m-%d')
+    request = {
+        'startDate': start_date,
+        'endDate': end_date,
+        # uneditable to enforce a nice clean dataframe at the end!
```
Comment (maxious) [inline google_searchconsole/gsc_bq_dta_dga.py:36]: what does this mean?

File: .gitignore [hunk 3]
```diff
@@ -164,3 +164,7 @@ Staticfile.auth
 *.ipr
 *.iws
 notebooks/*.html
+*.json
+google_searchconsole/parameters.py
```
Comment (maxious) [inline .gitignore:168]: Where is this used? If it is secret values, there should be a template file that isn't ignored.

File: google_searchconsole/gsc_bq_dta_dga.py [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+import re
+
+import pandas as pd
+
+from google.cloud import bigquery
+from google.oauth2 import service_account
+from googleapiclient.discovery import build
+from oauth2client.service_account import ServiceAccountCredentials
+
+import parameters as pe
```
Comment (maxious) [inline google_searchconsole/gsc_bq_dta_dga.py:11]: As above, provide a template for parameters. 

SERVICE_ACCOUNT_FILE might be better entered as a command line parameter or environment variable with a default value of "credentials.json" or something like that if it is not provided.

**Review by maxious — CHANGES_REQUESTED** [review]

File: google_searchconsole/gsc_bq_dta_dga.py (hunk 2 above)
Comment (mufaddaltaiyeb) [reply google_searchconsole/gsc_bq_dta_dga.py:36]: Remove the comment.

File: google_searchconsole/gsc_bq_dta_dga.py (hunk 4 above)
Comment (mufaddaltaiyeb) [reply google_searchconsole/gsc_bq_dta_dga.py:11]: parameter template file created for upload

File: .vscode/settings.json (hunk 1 above)
Comment (mufaddaltaiyeb) [reply .vscode/settings.json:6]: ok

File: .gitignore (hunk 3 above)
Comment (mufaddaltaiyeb) [reply .gitignore:168]: created template file for upload

**Review by sukhrajghuman — COMMENTED** [review]
Hey @mufaddaltaiyeb

Nice work!!

I think it would be nice to attribute the original code somewhere in the `gsc_bq_dta_dga.py` file.

https://github.com/benpowis/search-console-bq/blob/master/search_console_bq.py

It's good open source ettiquette :)

**Pull request merged**